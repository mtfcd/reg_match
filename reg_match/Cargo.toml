[package]
name = "reg_match"
description = "A match style regex tool."
version.workspace = true
edition.workspace = true

[dependencies]
regex = { workspace = true }
reg_match_macro = { path = "../reg_match_macro" }

[dev-dependencies]
trybuild = "1.0.101"
