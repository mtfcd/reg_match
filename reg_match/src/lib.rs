pub use reg_match_macro::reg_match;
pub use regex;
