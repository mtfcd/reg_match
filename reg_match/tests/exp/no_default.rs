use reg_match::reg_match;

fn main() {
    let e = "123abc";
    let a = reg_match!(e {
        "4" => "4".to_string(),
        r"(?<b>\d+(?<c>.+))" => format!("{}-{}", b, c),
        //_=>"a".to_string()
    });
    assert_eq!("123abc-abc", a);
}
