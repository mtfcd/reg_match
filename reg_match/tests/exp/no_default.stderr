error: proc macro panicked
 --> tests/exp/no_default.rs:5:13
  |
5 |       let a = reg_match!(e {
  |  _____________^
6 | |         "4" => "4".to_string(),
7 | |         r"(?<b>\d+(?<c>.+))" => format!("{}-{}", b, c),
8 | |         //_=>"a".to_string()
9 | |     });
  | |______^
  |
  = help: message: should always has a default branch.
